lalrpop/src/grammar/repr.rs

```diff
@@ -505,7 +505,7 @@ impl UserActionFnDefn {
                self.arg_patterns
                    .iter()
                    .zip(self.arg_types.iter())
-                   .map(|(p, t)| format!("{}: {}", p, t))
+                   .map(|(p, t)| format!("{}{}: {}", if p.0 { "mut " } else { "" }, p.1, t))
```

Comment:
I like the idea of having `p` be a pattern, or something that implements `Display`, so that here we can just leave this code unchanged (also the spot above), and `p` will just emit `mut x`  or `x` as appropriate.